tangelo/linq/target/target_cirq.py

```diff
@@ -110,13 +110,36 @@ def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, in
 
         # Run shot by shot and keep track of desired_meas_result only (generally slower)
         elif desired_meas_result or (save_mid_circuit_meas and return_statevector):
-            translated_circuit = translate_c(source_circuit, "cirq", output_options={"noise_model": self._noise_model,
-                                                                                     "save_measurements": True})
-            self._current_state = None
-            indices = list(range(source_circuit.width))
-            n_attempts = 0
+
+            if self.n_shots is None:
```

Comment:
I do not understand this line, I think L112 makes the cases overall very entangled.
The comments are essential here because all the different situations are hard to identify.

I'm not sure how the number of shots dictates whether or not you're gonna split the circuit (didn't seem to bother you in the 2 other files, but the noise_model was another story).

Would you please try to go through this and either add comments or refactor the conditional branches to make things a bit more clearer if you can figure out something ? Even with the full code file open, it is difficult for me to review so I imagine other readers will struggle as well.

Reply:
Added comments